core/src/main/java/org/mule/runtime/core/internal/management/stats/PayloadStatisticsCursorStream.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.core.internal.management.stats;
+
+import java.io.IOException;
+import java.util.function.LongConsumer;
+
+import org.mule.runtime.api.streaming.CursorProvider;
+import org.mule.runtime.api.streaming.bytes.CursorStream;
+
+final class PayloadStatisticsCursorStream extends CursorStream {
```

Comment:
why is this needed intead of using the inputstream decorator?

Reply:
same comment as before.